core/roboconf-messaging/src/main/java/net/roboconf/messaging/internal/client/rabbitmq/RabbitMqClientDm.java

```diff
@@ -219,6 +229,76 @@ public void listenToAgentMessages( Application application, ListenerCommand comm
 	}
 
 
+	@Override
+	public void sendMessageToDebug( MsgEcho echo, long ttl ) throws IOException {
+		this.logger.fine( "The DM sends a debug message to " + DEBUG_QUEUE_NAME + ". Message: " + echo.getContent() );
+
+		this.channel.basicPublish(
+				"",                                             // no exchange
+				DEBUG_QUEUE_NAME,                               // the queue
+				new BasicProperties.Builder()                   // Message properties (ttl)
+						.expiration( Long.toString( ttl ) )
+						.build(),
+				SerializationUtils.serializeObject( echo ) );    // Message content
+
+		this.logger.fine( "The DM sent a debug message to " + DEBUG_QUEUE_NAME + ". Message: " + echo.getContent() );
+	}
+
+
+	@Override
+	public void listenToDebugMessages( ListenerCommand command ) throws IOException {
+		switch (command) {
```

Comment:
3 days of hollidays, and I do not remember the way the messaging works.
Do we have one listening thread per application? I will take a look at it on tomorrow.